packages/web-pkg/src/composables/pagination/usePagination.ts

```diff
@@ -40,6 +45,19 @@ export function usePagination<T>(options: PaginationOptions<T>): PaginationResul
     return unref(options.items).slice(start, end)
   })
 
+  eventBus.subscribe(
+    'app.files.navigate.page',
+    ({ resourceId, forceScroll }: { resourceId: string; forceScroll: boolean }) => {
+      const index = findIndex(unref(options.items), (item: any) => item.id === resourceId)
+      if (index >= 0) {
+        const page = Math.ceil((index + 1) / Number(unref(perPage)))
+        router.push({ ...unref(route), query: { ...unref(route).query, page } }).then(() => {
+          eventBus.publish('app.files.navigate.scrollTo', { resourceId, forceScroll })
+        })
```

Comment:
We're getting somewhere! 

I'm not completely happy with this, but I played around with it and it's tricky to solve it directly through the page ref. I would like to try a few more things, but have to focus on other things for now. 
If that's okay for you, I will try again tonight and if I don't find anything useful, I'm fine to have this merged tomorrow.

Reply:
I obviously didn't  get to it, feel free to merge